code/modules/admin/permissionedit.dm

```diff
@@ -163,21 +417,21 @@ ADMIN_VERB(edit_admin_permissions, R_PERMISSIONS, "Permissions Panel", "Edit adm
 	if(CONFIG_GET(flag/load_legacy_ranks_only) && (task == "add" || task == "rank" || task == "permissions"))
 		to_chat(usr, "<span class='admin prefix'>Database rank loading is disabled, only temporary changes can be made to a rank's permissions and permanently creating a new rank is blocked.</span>", confidential = TRUE)
 		legacy_only = TRUE
-	if(check_rights(R_DBRANKS, FALSE))
-		if(!skip)
-			if(!SSdbcore.Connect())
-				to_chat(usr, span_danger("Unable to connect to database, changes are temporary only."), confidential = TRUE)
-				use_db = FALSE
-			else
-				use_db = tgui_alert(usr,"Permanent changes are saved to the database for future rounds, temporary changes will affect only the current round", "Permanent or Temporary?", list("Permanent", "Temporary", "Cancel"))
-				if(use_db == "Cancel")
-					return
-				if(use_db == "Permanent")
-					use_db = TRUE
-				else
-					use_db = FALSE
-			if(QDELETED(usr))
+
+	if(check_rights(R_DBRANKS, FALSE) && !skip)
+		if(!SSdbcore.Connect())
+			to_chat(usr, span_danger("Unable to connect to database, changes are temporary only."), confidential = TRUE)
+			use_db = FALSE
+		else
+			use_db = tgui_alert(usr,"Permanent changes are saved to the database for future rounds, temporary changes will affect only the current round", "Permanent or Temporary?", list("Permanent", "Temporary", "Cancel"))
+			if(use_db == "Cancel")
 				return
```

Comment:
```suggestion
			use_db = tgui_alert(usr,"Permanent changes are saved to the database for future rounds, temporary changes will affect only the current round", "Permanent or Temporary?", list("Permanent", "Temporary", "Cancel"))
			if(isnull(use_db) || use_db == "Cancel")
				return
```

Cancel and closing the window should be synonymous imo

Reply:
yea